### PR TITLE
feat: Enable Algolia Autocomplete search bar

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
This PR should allow for searching the articles on fCC News.

Just a few notes:
- A search page needs to be enabled. Just add the following to `content/settings/routes.yaml` in the Ghost install before restarting Ghost:
```
routes:
  /search/:
    template: search-results
```
- In `partials/algolia-search.hbs`, change `{ hint: false }` to `{ hint: false, debug: true }` so styling the suggestions in the search drop down easier. The default behavior is to close the dropdown whenever click outside the search bar/drop down.
- We might want to add the Algolia flies in `default.hbs` to the fCC CDN.

### CODE REVIEW

File: default.hbs [hunk 1]
```diff
@@ -20,6 +20,11 @@
 
     {{!-- Styles'n'Scripts --}}
     <link rel="stylesheet" type="text/css" href="{{asset "built/screen.css"}}" />
+    <link rel="stylesheet" type="text/css" href="{{asset "css/algolia-search.css"}}" />
+    
+    {{!-- Algolia search assets --}}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/algoliasearch/3.33.0/algoliasearch.min.js"></script>
```
Comment (raisedadead) [inline default.hbs]: Let's move these to the cdn.

File: partials/algolia-search.hbs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        empty: (options) => {
+          return `
+            <div class="algolia-result">
+              No tutorials found
+            </div>
+          `;
+        },
+        footer: (query, result) => {
+          if (!query.isEmpty) {
+            // CHANGE URL LATER! => https://freecodecamp.org/news/search?query=${result.query}
```
Comment (raisedadead) [inline partials/algolia-search.hbs]: ### 🚨 REMOVE AFTER QA 🚨

File: default.hbs (hunk 1 above)
Comment (scissorsneedfoodtoo) [reply default.hbs]: Sure thing, I'll open up a PR to the cdn and update this once it's merged.

**Comment by raisedadead** [comment]
Let me know when you are able to squash and then rebase this for a QA.

**Review by ahmaxed — APPROVED** [review]

**Comment by ahmaxed** [comment]
@raisedadead Tested locally, LGMT.
it works great on desktop and mobile.
@scissorsneedfoodtoo great work. the load more button works great on the search page.

**Comment by scissorsneedfoodtoo** [comment]
@ahmadabdolsaheb, thanks for your review, and all your help with deciding what to index and styling the search results page. I'll change the links to point to fCC News now. Then all we need to do is update `content/settings/routes.yaml` over on Ghost before deploying.

File: partials/algolia-search.hbs (hunk 2 above)
Comment (raisedadead) [reply partials/algolia-search.hbs]: Let get this resolved.

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by scissorsneedfoodtoo** [comment]
@raisedadead, just switched those links out. Should be good to go now.

**Comment by raisedadead** [comment]
Perfect, are we able to track the routes in the ghost-config repo, just for sanity?

**Review by raisedadead — APPROVED** [review]

**Comment by ahmaxed** [comment]
for reference this is how I got it setup locally 
```
routes:
  /rss/:
    template: rss
    content_type: text/xml
  /search/:
    template: search-results

collections:
  /:
    permalink: /{slug}/
    template: index

taxonomies:
  tag: /tag/{slug}/
  author: /author/{slug}/
```

**Pull request merged**